CRAW-216: Change the way of using Jsoup CSS Selectors

abhishekvm opened the pull request:
1. Replaced element[class=someclass] to element.classname
2. Replaced *[class=someclass] and [class=someclass] to .classname
3. Replaced \*[class\*=someclass] to [class*=someclass]
   - The starting * in \*[class\*=someclass] is redundant and does not create any differnce
4. Replaced *[id=someid] to [id=someid]
5. Replaced \*[id\*=someid] to [id*=someid]

We were expected to have some performance improvement as mentioned on https://airpr1.atlassian.net/browse/CRAW-216. After the change, I cross checked the performance and don't see any difference. Extractor takes approximately the same time before and after the change.

Commit:
1. Replaced element[class=someclass] to element.classname
2. Replaced *[class=someclass] and [class=someclass] to .classname
3. Replaced *[class*=someclass] to [class*=someclass]
   - The starting * in *[class*=someclass] is redundant and does not create any differnce
4. Replaced *[id=someid] to [id=someid]
5. Replaced *[id*=someid] to [id*=someid]

skyshard commented on src/main/java/de/jetwick/snacktory/ArticleTextExtractor.java:
```diff
@@ -133,66 +133,66 @@ public boolean hasHTMLTags(String text){
+                ".m-linkset__entries-item",
+                ".m-linkset",
+                ".feature-photos-story.feature-photos-column",
+                "[class*=js-carousel-pane]",
```
are we intentionally matching on partial class names or would this just be equivalent to `.js-carousel-pane`?

abhishekvm replied:
Need to check this out may be by changing to .classname and running testcases.

Commit:
Merge branch 'master' of https://github.com/skyshard/snacktory into abhishek/performance_improvement_css_selectors

Commit:
- Wherever possible changed [class*=someclass] to .someclass
- At few places it makes sense to keep more generic match with [class*=someclass]

andresp99999 reviewed (approved):
LGTM, even if performance difference is small, I think rules are easier to read this way.

andresp99999 commented:
@skyshard, are you ok with merging this branch?

Commit:
Merge branch 'master' of https://github.com/skyshard/snacktory into abhishek/performance_improvement_css_selectors

skyshard commented:
LGTM overall. There seems to be many cases of `[class*=something]` still where I'm not sure that we actually intend that behavior (wildcard matching anywhere within the class attribute, vs just matching any of the individual classes). We can address that in a different pull
